### PR TITLE
Remove 'Unreleased' section of CHANGE-NOTES.md

### PR DESCRIPTION
**Pull request by MaciejG604**

### CODE REVIEW

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/IntellijPlugin.kt [hunk 1]
```diff
@@ -100,5 +119,8 @@ fun Project.configureIntellijPlugin() {
     password.set(pluginSignPrivateKeyPass)
   }
 
-  tasks.withType<PublishPluginTask> { token.set(jetbrainsMarketplaceToken) }
+  tasks.withType<PublishPluginTask> {
+    dependsOn(verifyChangeLogTask)
```
Comment (PawelLipski) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/IntellijPlugin.kt:123]: This is a veery late moment to run this task... I'd suggest the following instead:
1. the `prospectiveVersionSection.version != latestVersionSection.version` check can be extracted to a separate tsk and made a dependency of `buildPlugin` task maybe?
2. the check for `prospectiveVersionSection` contents could also be extracted, and then executed from `scripts/enforce-change-notes-updated-on-pr-to-master` (to replace the existing logic there)
3. the existing `verifyChangeLog` could be retained (and depend on both tasks from 1. & 2.)... but only for manual use (not to be run automatically from CI or whatnot)

WDYT?

Comment (MaciejG604) [reply buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/IntellijPlugin.kt:123]: Yeah, that's reasonable 👍

File: scripts/enforce-change-notes-updated-on-pr-to-master [hunk 2]
```diff
@@ -15,12 +15,11 @@ if [[ ${CI_PULL_REQUEST-} && ${GITHUB_TOKEN-} ]]; then
...
-    fi
-    change_notes=$(./gradlew getChangelog -q)
-    if ! grep '^- ..*' <<< "$change_notes" &> /dev/null; then
-      die "$(head --lines=1 <<< "$change_notes" | sed 's/## //') section of $change_notes_path is empty or ill-formatted"
+    task_output=$(./gradlew verifyChangeLogContents)
+    fail=$(grep 'Execution failed' <<<"$task_output")
+
+    if [[ -n $fail ]]; then
+      die $(grep '> java.lang.Exception: ' <<<"$task_output" | sed "s/> java.lang.Exception: //")
     fi
```
Comment (PawelLipski) [inline scripts/enforce-change-notes-updated-on-pr-to-master:23]: Probably just
```suggestion
    ./gradlew verifyChangeLogContents
```
should be enough

File: buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/IntellijPlugin.kt [hunk 3]
```diff
@@ -35,18 +36,59 @@ fun Project.configureIntellijPlugin() {
...
   }
 
   val changelog = extensions.getByType(ChangelogPluginExtension::class.java)
 
+  val verifyStructureTask = tasks.register("verifyChangeLogStructure") {
+    doLast {
+      val prospectiveVersionSection = changelog.get(changelog.version.get())
+      val latestVersionSection = changelog.getLatest()
+
+      if (prospectiveVersionSection.version != latestVersionSection.version) {
```
Comment (PawelLipski) [inline buildSrc/src/main/kotlin/com/virtuslab/gitmachete/buildsrc/IntellijPlugin.kt:56]: Whoops I changed latest the version in CHANGE-NOTES.md from v3.2.0 to v3.2.1 and run `./gradlew verifyChangeLog`... got:

```

FAILURE: Build failed with an exception.

* Where:
Build file '/home/plipski/git-machete-intellij-plugin/build.gradle.kts' line: 228

* What went wrong:
Version is missing: [Unreleased]
> Version is missing: [Unreleased]

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
15 actionable tasks: 15 up-to-date
```

which is an unexpected error 🤔

File: scripts/enforce-change-notes-updated-on-pr-to-master [hunk 4]
```diff
@@ -15,12 +15,6 @@ if [[ ${CI_PULL_REQUEST-} && ${GITHUB_TOKEN-} ]]; then
...
   # PRs to master are typically either release or hotfix PRs, both of which should have change notes updated
   if [[ $base_branch == master ]]; then
-    if ! ./gradlew getChangelog -q --no-header &> /dev/null; then
-      die "Content of $change_notes_path has not been patched, fill the 'Unreleased' section and run ./gradlew patchChangelog"
-    fi
-    change_notes=$(./gradlew getChangelog -q)
-    if ! grep '^- ..*' <<< "$change_notes" &> /dev/null; then
-      die "$(head --lines=1 <<< "$change_notes" | sed 's/## //') section of $change_notes_path is empty or ill-formatted"
-    fi
+    ./gradlew verifyChangeLogContents > /dev/null
```
Comment (PawelLipski) [inline scripts/enforce-change-notes-updated-on-pr-to-master:18]:
```suggestion
    ./gradlew -q verifyChangeLogContents
```

**Review by PawelLipski — APPROVED** [review]

File: scripts/enforce-change-notes-updated-on-pr-to-master (hunk 4 above)
Comment (MaciejG604) [reply scripts/enforce-change-notes-updated-on-pr-to-master:18]: I wanted to keep the output of the script clean on success, but i guess displaying the progress bar for 3 seconds won't hurt.

**Pull request merged**